### ISSUE TITLE
Create Default Form When Pub Type Is Created

### ISSUE THREAD

**Issue by gabestein** [pubpub/platform#782]
## Motivation
So admins have a form to edit when they create a new type.
So devs have a form to render when they create a new type.

## Requirements
- When a new Pub Type is created, an accompanying new default Form for the type is created, following the current heuristics for the default form.
- Default form has name “<Pub Type> Editor (Default)”

## Acceptance Criteria
- Should specify the same form input types given field schema types as the current default form editor
- Should use the title of the field, not the slug, as the “Label” for each form field.
- Should add a paragraph field at the top with the title of the Pub (or fallback) rendered as text (not markdown, for now).
- Should add a migration to create default forms for existing pub types

**Comment by kalilsn** [comment]
>Should add a paragraph field at the top with the title of the Pub (or fallback) as text (not markdown, for now).

I don't really know how this is possible. If we're not adding a new markdown directive to render the pub's title value, I assume you mean we would statically set the contents of the paragraph to some pub title. But there's no pub when we're creating the default form, just a pub type.

We could achieve something like this by modifying the edit form (not create though) to always render the pub's title at the top

**Comment by gabestein** [comment]
oops — meant a Pub Type title, not Pub title.

**Comment by tefkah** [comment]
Right. Ideally we would then want a new markdown directive that renders the title, rather than a `value` directive for the field that is the title at time of creation. otherwise if you later change which field is title of the pubtype you would need to update the form again. Maybe that's fine for now, but it would be nicer to have it specifically pick out the title.

**Comment by tefkah** [comment]
> Should use the title of the field, not the slug, as the “Label” for each form field.

I agree with this, when did we actually change this? I think this should just be the default behavior for forms in general.

**Comment by kalilsn** [comment]
Makes sense! I don't think we can easily have a fallback value, but I'm assuming it's ok to just render nothing on the creation form?

**Comment by tefkah** [comment]
No fallback is probably fine for now, we can probably word it in a way where it does not look that weird if there's no actual title

**Comment by tefkah** [comment]
Can a default form be deleted? I think you should not be able to, unless the pubtype is deleted.

Additional requirement (possibly):
- add a column to the form table called `default`. set a unique index such that only one table can be default per pubtype.

**Comment by kalilsn** [comment]
>Additional requirement (possibly):
>
> - add a column to the form table called default. set a unique index such that only one table can be default per pubtype.


I think adding the column should be part of #781 most likely

**Comment by tefkah** [comment]
ah good point!

**Comment by kalilsn** [comment]
>Can a default form be deleted? I think you should not be able to, unless the pubtype is deleted.

Forms can't be deleted in general, only archived, which doesn't do anything. I don't think we need to worry about that until we make archiving actually do something. 

I would like to say that we should delete the form if the pub type is deleted (on cascade delete) but in the past we decided not to let forms be deleted or slugs modified to prevent dead pubpub links out in the wild (emails, bookmarks, etc.). Is that a concern here? Should we prevent the default form from being used in a public facing way so that it's safe to delete?

**Comment by gabestein** [comment]
> Should we prevent the default form from being used in a public facing way so that it's safe to delete?

No, I think we should allow the default form from being used in public facing ways. I think in a lot of cases people will use the same form in invites and internally. So I think we can continue to preserve existing form behavior.

**Comment by kalilsn** [comment]
Ok, just to be clear, the current behavior during pub type deletion is that you get an unexpected error if you delete a pub type that's attached to a form. So adding default forms without setting up cascading deletes from types -> forms will make it impossible to delete a pub type.

**Comment by gabestein** [comment]
Understood, and okay for now. We definitely need to resolve this — the reason I'm hesitant to now is my guess is it'll involve modifying the way we handle pub type deletion rather than form deletion.
